### PR TITLE
fix: version check

### LINKED ISSUE #872: Specify minimum required task verison.
While the `version:` key is nice to specify the Taskfile schema version, it would be nice to have some option to define a _minimum_ required task version to run a task.

For example, a recent change introduced `.ROOT_DIR`, which is incredibly handy for how I leverage task for my team, but many of my devs are stumbling over that as they haven't bothered to upgrade task yet.

I propopse that in in taskfiles, the `version:` key be expanded to check for minimum task version.

Example:

```
version: 3
``` 

would be compatible with all 3.x taskfile releases, but...

```
version: 3.15.2
```

would cause `task` to exit non-zero if the task version isn't _at least_ 3.15.2.

### PR DESCRIPTION
**Pull request by vmaerten**
Fixes #872

Following this PR : 
- https://github.com/go-task/task/pull/1465

https://github.com/go-task/task/blob/a74b0bc679e770179c43e265f134feaa3b1b733e/setup.go#L241-L253
This does not work when the sum is displayed in the version
Cf : 
image
And
https://github.com/go-task/task/blob/a74b0bc679e770179c43e265f134feaa3b1b733e/internal/version/version.go#L21-L23

### CODE REVIEW

File: internal/version/version.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 
 var version = ""
```
Comment (pd93) [inline internal/version/version.go:8]: I don't think this variable is currently set/used?

Might be nice to do something like the following:

```go
var (
    version = ""
    sum = ""
)

func init() {
  	info, ok := debug.ReadBuildInfo()
	if !ok || info.Main.Version == "" {
		version = "unknown"
	}
    version = info.Main.Version
	sum = info.Main.Sum
}
```

This way, we're not fetching the version info every time `GetVersion` is called.

File: internal/version/version.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
...
 
-func GetVersion() string {
+func GetVersion(withSum bool) string {
```
Comment (pd93) [inline internal/version/version.go:10]: If following the previous suggestion, I think I would also prefer two separate functions. This feels more Go-like and readable in calling code to me than parameter flags:

```go
func GetVersion() string {
    return version
}

func GetVersionWithSum() string {
    return fmt.Sprintf("%s (%s)", version, sum)
}
```

**Review by pd93 — COMMENTED** [review]
Thanks for fixing this. Bit of an oversight on my part when doing #1465. A couple of minor suggestions

File: internal/version/version.go (hunk 2 above)
Comment (vmaerten) [reply internal/version/version.go:10]: I had the same question when doing this.
Good to me :)

File: internal/version/version.go (hunk 1 above)
Comment (vmaerten) [reply internal/version/version.go]: This way it's cleaner, thanks!

**Comment by vmaerten** [comment]
It's done @pd93 
Thanks a lot for your review, especially about "Go-like code" Even if we are using Go at work, it's always good to know 
good practices

**Review by pd93 — APPROVED** [review]
Looks good to me. Thanks!

File: internal/version/version.go (hunk 1 above)
Comment (stefanb) [reply internal/version/version.go]: Please note that building from source tarball without any git info still requires setting the version via ldflags.
See
* https://github.com/go-task/task/issues/1709

**Pull request merged**